src/app/dashboard/characters/attack-editor-dialog/attack-editor-dialog.component.ts

```diff
@@ -64,4 +67,47 @@ export class AttackEditorDialog implements OnInit {
         }
       });
   }
+
+  // JSON
+  beginJSONExport(attack: Attack | Attack[]) {
+    this.dialog.open(JSONExportDialog, {
+      data: {name: (attack instanceof Array) ? 'All Attacks' : attack.name, data: attack},
+      width: '60%'
+    });
+  }
+
+  beginJSONImport() {
+    const dialogRef = this.dialog.open(JSONImportDialog, {
+      width: '60%',
+      disableClose: true,
+      data: {validator: (data) => this.validateAttackJSON(dialogRef, data)}
+    });
+
+    dialogRef.afterClosed().subscribe(result => {
+      if (result) {
+        const attack: Attack | Attack[] = result;
+        console.log(attack);
```

Comment:
Did you mean to commit this console log?

Reply:
Not exactly yes or no - is it bad to have verbose logging like this? Usually I just leave logs like this in since then Sentry can pick it up or some more technical users can look at it

Reply:
It's not _bad_ per se, it's just generally considered somewhat unprofessional to have a production build logging stuff into the console during normal usage. I personally find it gets difficult to find logs I actually need during development when I allow myself to keep older logs in source code, because eventually the volume of "happy path" console logs will drown out any other logs that are added for one-off testing.

If you find value in these logs though, especially if they are being helpful for sentry log debugging, then by all means keep them. If this is the main use case you anticipate for these logs though, I would suggest adding some unique searchable key that ties it to this bit of code might be helpful if that's the goal (e.g. `console.log('attack-editor-dialog closed', attack)`). I do that for the explicit sentry error logs I add (usually API unknown error handling) so I can easily find the exact call point in the code that could trigger an error that I had the forethought to write logging code for.

Reply:
Fair enough - maybe I'm just used to sifting through logs on sites like R20 😛